Refactor routing configuration using TypeScript

furinvader opened the pull request.

furinvader committed:
Move routing to components

furinvader committed:
Remove annotations in routing configuration

furinvader committed:
Use $window service instead of plain window object in configuration

furinvader committed:
Replace function keyword with arrow syntax

furinvader committed:
Use type annotations for injected modules

Also add the interfaces being extended in the code.
Some of those are private properties made public.

furinvader committed:
Refactor the run configuration

furinvader committed:
Add relations panel declaration to panels object

furinvader committed:
Move routing file to config components

TeatroIO commented:
I've prepared a stage to preview changes. <a href='http://furinvader-refactoring-routing.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/' target='_blank'>Open stage</a> or <a href='http://furinvader-refactoring-routing.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com//teatro.log' target='_blank'>view logs</a>.

furinvader committed:
Move routing test file to components

furinvader committed:
Add tsd npm module to dependencies

Install it globally, as it seems to be the only way the package
gets installed without failing.
